wcfsetup/install/files/ts/WoltLabSuite/Core/Ui/Dropdown/Simple.ts

```diff
@@ -0,0 +1,596 @@
+          if (!listItem.clientHeight) return;
+          if (firstListItem === null) firstListItem = listItem;
+          else if (listItem.classList.contains('active')) firstListItem = listItem;
+
+          listItem.setAttribute('role', 'menuitem');
+          listItem.tabIndex = -1;
+        });
+      }
+
+      UiDropdownSimple.setAlignment(dropdown, menu, alternateElement);
+
+      if (firstListItem !== null) {
+        firstListItem.focus();
+      }
+    }
+  });
+
+  window.WCF.Dropdown.Interactive.Handler.closeAll();
+
+  return (event === null);
+}
+
+function handleKeyDown(event: KeyboardEvent): void {
+  // <input> elements are not valid targets for drop-down menus. However, some developers
+  // might still decide to combine them, in which case we try not to break things even more.
+  const target = event.currentTarget as HTMLElement;
+  if (target.nodeName === 'INPUT') {
+    return;
+  }
+
+  if (event.key === 'Enter' || event.key === 'Space') {
+    event.preventDefault();
+    toggle(event);
+  }
+}
+
+function dropdownMenuKeyDown(event: KeyboardEvent): void {
+  let button, dropdown;
+
+  const activeItem = document.activeElement as HTMLElement;
+  if (activeItem.nodeName !== 'LI') {
+    return;
+  }
+
+  if (event.key === 'ArrowDown' || event.key === 'ArrowUp' || event.key === 'End' || event.key === 'Home') {
+    event.preventDefault();
+
+    const listItems = Array.from<HTMLElement>(activeItem.closest('.dropdownMenu')!.querySelectorAll('li'));
+    if (event.key === 'ArrowUp' || event.key === 'End') {
+      listItems.reverse();
+    }
+
+    let newActiveItem: HTMLElement | null = null;
+    const isValidItem = listItem => {
+      return !listItem.classList.contains('dropdownDivider') && listItem.clientHeight > 0;
+    };
+
+    let activeIndex = listItems.indexOf(activeItem);
+    if (event.key === 'End' || event.key === 'Home') {
+      activeIndex = -1;
+    }
+
+    for (let i = activeIndex + 1; i < listItems.length; i++) {
+      if (isValidItem(listItems[i])) {
+        newActiveItem = listItems[i];
+        break;
+      }
+    }
+
+    if (newActiveItem === null) {
+      for (let i = 0; i < listItems.length; i++) {
+        if (isValidItem(listItems[i])) {
+          newActiveItem = listItems[i];
+          break;
+        }
+      }
+    }
+
+    if (newActiveItem !== null) {
+      newActiveItem.focus();
+    }
+  } else if (event.key === 'Enter' || event.key === 'Space') {
+    event.preventDefault();
+
+    let target = activeItem;
+    if (target.childElementCount === 1 && (target.children[0].nodeName === 'SPAN' || target.children[0].nodeName === 'A')) {
+      target = target.children[0] as HTMLElement;
+    }
+
+    dropdown = _dropdowns.get(_activeTargetId);
+    button = dropdown.querySelector('.dropdownToggle');
+
+    const mouseEvent = dropdown.dataset.a11yMouseEvent || 'click';
+    Core.triggerEvent(target, mouseEvent);
+
+    if (button) button.focus();
```

Comment:
```suggestion
    if (button) {
      button.focus();
    }
```
(same below)

Reason: Braces make the code more readable if more than a simple `return` happens.